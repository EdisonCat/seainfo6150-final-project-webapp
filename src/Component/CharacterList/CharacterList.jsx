import React, { useEffect, useState } from "react";
import styles from "./CharacterList.module.css";
import CharacterCard from "../CharacterCard/CharacterCard.jsx";
export default function CharacterList() {
	const [items, setItems] = useState([]);
	const [page, setPage] = useState(0);
	const [loading, setLoading] = useState(true);
	useEffect(() => {
		setLoading(true);
		const fetchData = async () => {
			const url = `${process.env.REACT_APP_MARVEL_CHARACTERS_PATH}${process.env.REACT_APP_MARVEL_API_HASH}&offset=${page}`;
			const res = await fetch(url, {
				method: "GET",
				headers: { "Content-Type": "application/json" },
			});
			const json = await res.json();
			setItems(json.data.results);
			setLoading(false);
		};
		fetchData();
	}, [page]);
	return (
		<div className={styles.container}>
			{loading ? (
				"Loading..."
			) : (
				<div className={styles.list}>
					{items.map((item, idx) => (
						<div key={idx}>
							<CharacterCard data={item} />
						</div>
					))}
				</div>
			)}
			<div className={styles.cta}>
				{page === 0 ? null : (
					<button onClick={() => setPage(page - 20)}>Previous</button>
				)}
				<button onClick={() => setPage(page + 20)}>Next</button>
			</div>
		</div>
	);
}