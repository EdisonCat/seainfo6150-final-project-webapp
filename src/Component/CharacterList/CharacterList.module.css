.container {
    min-height: 900px;
    background-color: #333;
    border-top: 1px solid grey;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    color: white;
}
.list {
    display: grid;
    grid-template-columns: repeat(4, 1fr);
}
.cta {
    display: flex;
    align-items: center;
    justify-content: center;
}
.cta button {
    outline: none;
    background: white;
    color: black;
    transition: 0.5s;
    width: 100px;
}
.cta button:hover {
    background: red;
    color: white;
    width: 120px;
}
@media (max-width: 1200px) {
    .list {
        grid-template-columns: repeat(3, 1fr);
    }
}
@media (max-width: 800px) {
    .list {
        grid-template-columns: repeat(2, 1fr);
    }
}
@media (max-width: 700px) {
    .list {
        grid-template-columns: 1fr;
    }
}
