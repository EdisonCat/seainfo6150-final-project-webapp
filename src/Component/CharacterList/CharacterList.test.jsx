import React from "react";
import CharacterList from "./CharacterList";
it("rendered CharacterList correctly", () => {
	// eslint-disable-next-line no-undef
    const { container } = render(<CharacterList />);
    expect(container).toMatchSnapshot();
});