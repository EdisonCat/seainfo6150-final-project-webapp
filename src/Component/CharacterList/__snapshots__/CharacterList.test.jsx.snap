// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendered CharacterList correctly 1`] = `
<div>
  <div
    class="container"
  >
    Loading...
    <div
      class="cta"
    >
      <button>
        Next
      </button>
    </div>
  </div>
</div>
`;
